ngraph/python/tests/coverage/compliance/reporter.py

```diff
@@ -0,0 +1,83 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from .helpers import find_tested_op, UNSUPPORTED_OPS
+
+
+class ONNXTest:
+    def __init__(self, item):
+        self.name = item.name
+        self.mark = item.get_closest_marker("onnx_coverage")
+        self.result = ""
+        self.op = find_tested_op(item.nodeid)
+
+
+class ComplianceReporter:
+    def __init__(self):
+        self.test_info = {}
+        self.compliance = {}
+        self.test_report = []
+        self.ops_report = {}
+
+    def collect_test_info(self, item):
+        test_info = ONNXTest(item)
+        if test_info.mark.args[1] != "RealModel":
+            self.test_info[item.nodeid] = test_info
+
+    def add_test_result(self, nodeid, result):
+        if nodeid in self.test_info:
+            self.test_info[nodeid].result = result
+
+    def report_tests(self, report_path):
+        self.test_report.sort()
+
+        with open(report_path, "w") as report:
+            report.writelines(self.test_report)
+
+    def report_ops(self, report_path):
+        ops_report_lines = list(map(lambda op: op + ";Failed\n", UNSUPPORTED_OPS))
+
+        for op in self.ops_report:
+            op_test_results = self.ops_report[op]
+            if all(res == "passed" for res in op_test_results):
+                ops_report_lines.append(op + ";Passed\n")
+            elif all(res == "failed" for res in op_test_results):
+                ops_report_lines.append(op + ";Failed\n")
+            else:
+                ops_report_lines.append(op + ";Partial\n")
+
+        ops_report_lines.sort()
+        with open(report_path, "w") as report:
+            report.writelines(ops_report_lines)
+
+    def prepare_report_data(self):
+        for nodeid in self.test_info:
+            test = self.test_info[nodeid]
+            self.test_report.append(test.name + ";" + test.result + "\n")
+            self.add_test_result_to_op_statistics(test)
+
+    def add_test_result_to_op_statistics(self, test):
+        if test.mark.args[0] is not None:
+            model = self._model_from_mark(test.mark.args[0])
+            if model is None:
+                print("The test does not contain ModelProto:", test.name)
```

Comment:
These prints can pollute an output.
Maybe such tests should be ignored silently.

Reply:
Those lines don't show up in the current version. I've left those checks (and possible logs) for the future if there are any new tests that are weird or broken/incomplete. Currently there are only 9 test which fall into this category (no ModelProto in the pytest marker) but they are all skipped so we don't get even a single log file.

Reply:
Aditionally those logs are grouped and will only show up when the compliance report is enabled explicitly. Normal runs, both local and in CI, are not affected.